runtime/drivers/duckdb/model_manager.go

```diff
@@ -0,0 +1,176 @@
+package duckdb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type ModelInputProperties struct {
+	SQL string `mapstructure:"sql"`
+}
+
+func (p *ModelInputProperties) Validate() error {
+	if p.SQL == "" {
+		return fmt.Errorf("missing property 'sql'")
+	}
+	return nil
+}
+
+type ModelOutputProperties struct {
+	Table               string                      `mapstructure:"table"`
+	Materialize         *bool                       `mapstructure:"materialize"`
+	UniqueKey           []string                    `mapstructure:"unique_key"`
+	IncrementalStrategy drivers.IncrementalStrategy `mapstructure:"incremental_strategy"`
+}
+
+func (p *ModelOutputProperties) Validate(opts *drivers.ModelExecutorOptions) error {
+	if opts.Incremental {
+		if p.Materialize != nil && !*p.Materialize {
+			return fmt.Errorf("incremental models must be materialized")
+		}
+		p.Materialize = boolPtr(true)
+	}
+
+	if opts.InputConnector != opts.OutputConnector {
+		if p.Materialize != nil && !*p.Materialize {
+			return fmt.Errorf("models that output to a different connector must be materialized")
```

Comment:
**Issue**: Similar to the previous issue, the error message here lacks a detailed explanation which could help developers understand the constraints better. <br> **Fix**: Enhance the error message to include reasons why models outputting to a different connector require materialization. <br> **Code Suggestion**: 
 ```
 - return fmt.Errorf("models that output to a different connector must be materialized")
 + return fmt.Errorf("Models that output to a different connector must be materialized to manage data consistency across different storage systems.")
 ```